app/src/main/java/io/github/notefydadm/notefy/database/DatabaseHandler.java

```diff
@@ -0,0 +1,178 @@
+                        if (e != null) {
+                            callback.onFailureOnListener(e);
+                            return;
+                        }else{
+                            for (DocumentChange dc : snapshots.getDocumentChanges()) {
+                                Note note = getNoteFromQueryDocumentSnapshot(dc.getDocument());
+                                switch (dc.getType()) {
+                                    case ADDED:
+                                        callback.onNoteAdded(note);
+                                        break;
+                                    case MODIFIED:
+                                        callback.onNoteModified(note);
+                                        break;
+                                    case REMOVED:
+                                        callback.onNoteRemoved(note);
+                                        break;
+                                }
+                            }
+
+                        }
+
+                    }
+                });
+    }
+
+
+
+    private static Note getNoteFromQueryDocumentSnapshot (QueryDocumentSnapshot snapshot){
+        String id = snapshot.getId();
+        String tittle = snapshot.getString("tittle");
+        String userId = snapshot.getString("user_id");
+        NoteState state = getStateFromString(snapshot.getString("state"));
+        boolean isFavourite = snapshot.getBoolean("is_favourite");
+        int colour = Math.toIntExact(snapshot.getLong("colour"));
+        LocalDateTime creationDate = LocalDateTime.ofEpochSecond(snapshot.getTimestamp("creation_date").getSeconds(),0, ZoneOffset.UTC);
+        LocalDateTime lastModifiedDate = LocalDateTime.ofEpochSecond(snapshot.getTimestamp("last_modified_date").getSeconds(),0, ZoneOffset.UTC);
+
+
+        Note item = new Note(id,tittle,userId,state,isFavourite,colour,creationDate,lastModifiedDate,null);
+        return item;
+    }
+
+    private static NoteState getStateFromString(String stateInString){
+        switch (stateInString){
+            case("DRAFT"):
+                return DRAFT;
+            case("PUBLISHED"):
+                return PUBLISHED;
+            case("HIDDEN"):
+                return HIDDEN;
+            case("ARCHIVED"):
+                return ARCHIVED;
+            case("DELETED"):
+                return DELETED;
+            default:
+                return null;
+
+        }
+
+    }
+
+    private static String getStringFromState(NoteState state){
+        switch (state){
+            case DRAFT:
+                return "DRAFT";
+            case PUBLISHED:
+                return "PUBLISHED";
+            case HIDDEN:
+                return "HIDDEN";
+            case ARCHIVED:
+                return "ARCHIVED";
+            case DELETED:
+                return "DELETED";
+            default:
+                return "";
+
+        }
+
```

Comment:
Igual que en el comentario anterior.
```suggestion
        return state.toString();
```